hbase-client/src/main/java/org/apache/omid/transaction/SnapshotFilterImpl.java

```diff
@@ -298,26 +291,62 @@ public CommitTimestamp locateCellCommitTimestamp(long cellStartTimestamp, long e
         return commitCache;
     }
 
-    private void buildFamilyDeletionCache(List<Cell> rawCells, Map<String, List<Cell>> familyDeletionCache) {
-
+    private void buildFamilyDeletionCache(HBaseTransaction transaction, List<Cell> rawCells, Map<String, Long> familyDeletionCache, Map<Long, Long> commitCache, Map<String,byte[]> attributeMap) throws IOException {
         for (Cell cell : rawCells) {
             if (CellUtil.matchingQualifier(cell, CellUtils.FAMILY_DELETE_QUALIFIER) &&
                     CellUtil.matchingValue(cell, HConstants.EMPTY_BYTE_ARRAY)) {
-
                 String row = Bytes.toString(cell.getRow());
-                List<Cell> cells = familyDeletionCache.get(row);
-                if (cells == null) {
-                    cells = new ArrayList<>();
-                    familyDeletionCache.put(row, cells);
+                String family = Bytes.toString(cell.getFamily());
+                String key = row + ":" + family;
+
+                if (familyDeletionCache.containsKey(key))
+                    return;
+
+                Optional<Long> commitTimeStamp = isCellInSnapshot(cell, transaction, commitCache);
+
+                if (!commitTimeStamp.isPresent()) {
+                    commitTimeStamp = isCellInTransaction(cell, transaction, commitCache);
                 }
 
-                cells.add(cell);
+                if (commitTimeStamp.isPresent()) {
+                    familyDeletionCache.put(key, commitTimeStamp.get());
+                } else {
+                    Cell lastCell = cell;
+                    Map<Long, Long> cmtCache;
+                    boolean foundCommitttedFamilyDeletion = false;
+                    while (!foundCommitttedFamilyDeletion) {
+
+                        Get g = createPendingGet(lastCell, 3);
+                        for (Map.Entry<String,byte[]> entry : attributeMap.entrySet()) {
```

Comment:
Why do we need to update attributes? 
Its just a simple get we know exactly what we want and dont want server to filter anything

Reply:
I don't know which attribute the user wrote and whether he/she wrote a coprocessor that looks on these attributes and do something.

Reply:
But we know exactly what we want, a delete cell which is something to do only with omid.
you are adding attributes that where aimed for the get of a cell by the user, to a delete cell of omid